facebook/facebook-util.h

```diff
@@ -37,6 +37,38 @@
 #define FB_UTIL_ERROR  fb_util_error_quark()
 
 /**
+ * FB_UTIL_PTRBIT_GET:
+ * @p: Pointer used to store flags.
+ * @b: Bit to get from p.
+ *
+ * Get bit b from pointer p.
+ *
+ * Returns: Value of bit b in pointer p.
+ */
+#define FB_UTIL_PTRBIT_GET(p, b) \
+    (GPOINTER_TO_INT((p)) & (1 << ((b) - 1)))
+
+/**
+ * FB_UTIL_PTRBIT_SET:
+ * @p: Pointer used to store flags.
+ * @b: Bit to set in p.
+ * @v: Set or clear bit b
+ *
+ * Set bit b in pointer p based on value v
+ */
+#define FB_UTIL_PTRBIT_SET(p, b, v)     \
```

Comment:
Usually not a good idea to define variables in a macro of such, as it could result in a conflict. You could use something like `__tmp` or whatever, but there is no need for that here. Instead of conditionalizing things, just "reset" the bit by setting it to zero, then OR the shifted value (which may also be zero). Simply setting a bit is likely faster than branching anyways.